native/src/snappyclient/cpp/impl/SSLParameters.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * SSLParameters.h
+ *
+ *  Created on: 13-Dec-2019
+ *      Author: pbisen
+ */
+
+#ifndef SRC_SNAPPYCLIENT_HEADERS_SSLPARAMETERS_H_
+#define SRC_SNAPPYCLIENT_HEADERS_SSLPARAMETERS_H_
+
+#include "ClientBase.h"
+
+namespace io {
+  namespace snappydata {
+    namespace client {
+      namespace impl {
+
+        enum class SSLProperty
+          : int {
+            CIPHERSUITES = 0,
+          CLIENTAUTH = 1,
+          ENABLEPROTOCOLS = 2,
+          KEYSTORE = 3,
+          KEYSTOREPASSWORD = 4,
+          PROTOCOL = 5,
+          TRUSTSTORE = 6,
+          TRUSTSTOREPASSWORD = 7
+        };
+
+        class SSLParameters {
+        private:
+//          SSLParameters();
+//          ~SSLParameters();
+//        SSLParameters(const SSLParameters&) = delete; // no instances
+//        SSLParameters operator=(const SSLParameters&) = delete; // no instances
+          std::map<std::string, std::string> sslPropValMap;
+          std::set<std::string> sslProperties = { "cipher-suites",
+                                                  "client-auth",
+                                                  "enabled-protocols",
+                                                  "keystore",
+                                                  "keystore-password",
+                                                  "protocol", "truststore",
+                                                  "truststore-password" };
+        public:
```

Comment:
keystore-password and truststore-password fields are passed to OpenSSL API by function callback TSSLSocketFactory.getPassword. This needs to call the appropriate callback method to fetch password field from UI by extending TSSLSocketFactory. This can be tackled as a separate task and needs discussion on how to tie up UI with the callback cleanly (and also handle the case when UI is not being used directly).

Reply:
callback TSSLSocketFactory.getPassword has been added and password field in UI also added at DSN